optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/rotation/ShiftBlobView.java

```diff
@@ -0,0 +1,170 @@
+
+@Templated
+public class ShiftBlobView implements BlobView<Long, ShiftBlob> {
+
+    private static final Long BLOB_POSITION_DISPLACEMENT_IN_SCREEN_PIXELS = 3L;
+    private static final Long BLOB_SIZE_DISPLACEMENT_IN_SCREEN_PIXELS = -5L;
+
+    @Inject
+    @DataField("blob")
+    private HTMLDivElement root;
+
+    @Inject
+    @Named("span")
+    @DataField("label")
+    private HTMLElement label;
+
+    @Inject
+    private CircularDraggability<Long, ShiftBlob> draggability;
+
+    @Inject
+    private CircularResizability<Long, ShiftBlob> resizability;
+
+    private Viewport<Long> viewport;
+    private SubLane<Long> subLane;
+    private ListView<ShiftBlob> blobViews;
+    private Runnable onDestroy;
+
+    private ShiftBlob blob;
+
+    @Override
+    public ListElementView<ShiftBlob> setup(final ShiftBlob blob,
+                                            final ListView<ShiftBlob> blobViews) {
+
+        this.blob = blob;
+        this.blobViews = blobViews;
+
+        refresh();
+
+        draggability.applyFor(blob, BLOB_POSITION_DISPLACEMENT_IN_SCREEN_PIXELS, blobViews, subLane.getCollisionDetector(), viewport, this);
+        draggability.onDrag(this::onDrag);
+
+        resizability.applyFor(blob, BLOB_SIZE_DISPLACEMENT_IN_SCREEN_PIXELS, blobViews, subLane.getCollisionDetector(), viewport, this);
+        resizability.onResize(this::onResize);
+
+        return this;
+    }
+
+    private void refresh() {
+        positionBlobOnGrid();
+        updateLabel();
+    }
+
+    private void positionBlobOnGrid() {
+        viewport.setPositionInScreenPixels(this, blob.getPositionInGridPixels(), BLOB_POSITION_DISPLACEMENT_IN_SCREEN_PIXELS);
+        viewport.setSizeInScreenPixels(this, blob.getSizeInGridPixels(), BLOB_SIZE_DISPLACEMENT_IN_SCREEN_PIXELS);
+    }
+
+    private void updateLabel() {
+        //FIXME: Bad labeling
+        final String start = ((blob.getPositionInScaleUnits() / 60) + 24) % 24 + ":00";
+        final String end = ((blob.getEndPositionInScaleUnits() / 60) + 24) % 24 + ":00";
+        label.textContent = start + " to " + end;
+    }
+
+    private void onResize(final Long newSizeInGridPixels,
+                          final CollisionState dragState) {
+
+        refresh();
+        refreshTwinIfAny();
+
+        if (dragState.equals(COLLIDING)) {
+            DomGlobal.console.info("Collision!");
+        }
+    }
+
+    private void onDrag(final Long newPositionInGridPixels,
+                        final CollisionState dragState) {
+
+        refresh();
+        refreshTwinIfAny();
+
+        if (dragState.equals(COLLIDING)) {
+            DomGlobal.console.info("Collision!");
+        }
+    }
+
+    private void refreshTwinIfAny() {
+        blob.getTwin()
+                .map(blobViews::getView)
+                .map(view -> (ShiftBlobView) view)
+                .ifPresent(ShiftBlobView::refresh);
+    }
+
+    @EventHandler("blob")
+    public void onBlobClicked(final ClickEvent event) {
```

Comment:
Any reason for not registering the event handler with a native click event? I believe this should work:

    @EventHandler("blob")
    public void onBlobClicked(final @ForEvent("click") MouseEvent event) {

Reply:
It works. I remember trying it in the Workbench and it didn't work so probably it was missing the `@ForEvent` annotation when I tried